src/Compilers/CSharp/Portable/Parser/QuickScanner.cs

```diff
@@ -152,7 +149,7 @@ private enum CharFlags : byte
                 (byte)QuickScanState.FollowingCR,         // CR
                 (byte)QuickScanState.DoneAfterNext,       // LF
                 (byte)QuickScanState.Done,                // Letter
-                (byte)QuickScanState.Number,              // Digit
+                (byte)QuickScanState.Bad,                 // Dot followed by number.  Could be a fp `.0` or could be a range + num `..0`.  Can't tell here.
```

Comment:
this was not-fun to track down :D

the quick scanner is fairly simple.  but whne it's not in sync with teh actual scanner... le-sigh...

Reply:
Is there a specific test or assert we can add to make that obvious?

Reply:
yes.  that's what caught this :)

Reply:
Sorry, which test are you referring to?

Reply:
the ones that failed prior to this change being made :D

Reply:
the test caught this, which was great.  my general comment was simply that it's non trivial to know to go update this when updating the scanne.r  you do need to depend on the tests saying that something went wrong, causing you to go and figure out how to best update the quick scanner ot match the actual lexer.

Reply:
Right. I'm asking what we can do to make it more trivial to know how to update this. Is it better messaging? Another test?

Reply:
I think the test is reasonable.  It says that quick lexing produced different tokens.  If someone wanted to write something that maybe tried to go chra by char to figure out perhaps where they diverged, it might be ok. Not sure how viable that is.